src/components/organisms/AdminVenueView/components/EventsView/CsvImportModal.tsx

```diff
@@ -0,0 +1,219 @@
+            <div className="CsvImportModal__result-count">
+              The number of new events is {importResult.updateCount}.
+            </div>
+            {importResult.errors.length > 0 && (
+              <div className="CsvImportModal__result-errors">
+                Some events weren&apos;t addded due to invalid data. Please
+                check lines:
+                {" " +
+                  importResult.errors
+                    .map(({ index, reason }) => `#${index + 1} (${reason})`)
+                    .join(", ")}
+                .
+              </div>
+            )}
+          </div>
+        )}
+        {hasImportError && (
+          <div className="CsvImportModal__error CsvImportModal__error--import">
+            <span className="CsvImportModal__error-static">
+              There has been an error importing the data. Error:
+            </span>
+            <br />
+            <span className="CsvImportModal__error-generated">
+              {"" + importError?.message}
+            </span>
+          </div>
+        )}
+        {hasParseError && parsedErrors?.length === 1 && (
+          <div className="CsvImportModal__error CsvImportModal__error--single">
+            <span className="CsvImportModal__error-static">
+              There has been an error parsing the CSV file. Please make sure
+              everything is OK before proceeding. Error:
+            </span>
+            <br />
+            <span className="CsvImportModal__error-generated">
+              {parsedErrors[0].message}
+            </span>
+          </div>
+        )}
+        {hasParseError && parsedErrors && parsedErrors?.length > 1 && (
+          <div className="CsvImportModal__error CsvImportModal__error--multiple">
+            <span className="CsvImportModal__error-static">
+              There have been multiple errors parsing the CSV file. Please make
+              sure everything is OK before proceeding. Errors:
+            </span>
+            <br />
+            <span className="CsvImportModal__error-generated">
+              {parsedErrors.map((e) => e.message).join(". ")}
+            </span>
+          </div>
+        )}
+        <div className="CsvImportModal__wrapper">
+          {!parsedData && (
+            <span className="CsvImportModal__description">
+              The result of parsing the CSV file will be displayed here. You can
+              download an example to get you started
+            </span>
+          )}
+          {parsedData && (
+            <table className="CsvImportModal__table">
+              <tr>
+                <th>#</th>
+                {parsedMeta?.fields?.map((heading) => (
+                  <th key={heading}>{heading}</th>
+                ))}
+              </tr>
+              {parsedData.map((row, rowIndex) => (
+                <tr key={rowIndex}>
+                  <td>{rowIndex + 1}</td>
+                  {Object.values(row).map((value, cellIndex) => (
```

Comment:
How expensive can this get? If the answer is a lot, then add `useMemo` please.

Reply:
Can't predict every scenario. Some might even hang the parsing/whole UI (had accident picking Gimp image instead of CSV file). Will use memo.

Reply:
Added `useMemo` @ 15e9b6d4694e2433775d0d35b60a14ebbe68e847